src/components/inputs/text-input/text-input.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import invariant from 'tiny-invariant';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import Constraints from '../../constraints';
+import styles from './text-input.mod.css';
+
+type HorizontalConstraint = 's' | 'm' | 'l' | 'xl' | 'scale';
```

Comment:
Is this necessary? Can't we import it from the horizontal component?

Reply:
No, cause it's a subset of those props. Not every component supports every constraint. @montezume had a PR to go over all. We could import the type and `Exclude` some but I found that dangerous as given the HorizontalConstraint adds one it does not imply it should/needs to be automatically supported by this or another component.